cabal.haskell-ci

```diff
@@ -1,3 +1,12 @@
 branches: master ci*
 
 haddock: <8.1 || >8.3
+
+constraint-set text-2.1
+  ghc: >= 8.2
+  constraints: text ^>= 2.1
+  tests: True
+  run-tests: True
+
+raw-project
+  allow-newer: text
```

Comment:
What is this for? Is it saying to force text >= 2.1 when GHC >= 8.2 ? What happens when this isn't included?

Reply:
A `constraint-set` is another item in CI that tries to build with another configuration, in this case with `text >= 2.1`.  It is only enabled for GHC 8.2 or higher.  Since dependencies may have a `text < 2.1` bound, this is relaxed (`allow-newer`).